packages/trpc/server/routers/viewer/teams/hasActiveTeamPlan.handler.ts

```diff
@@ -22,18 +23,27 @@ export const hasActiveTeamPlanHandler = async ({ ctx }: HasActiveTeamPlanOptions
     },
   });
 
-  if (!teams.length) return false;
+  if (!teams.length) return { isActive: false, isTrial: false };
 
   // check if user has at least on membership with an active plan
   for (const team of teams) {
     const teamBillingService = new InternalTeamBilling(team);
     const isPlanActive = await teamBillingService.checkIfTeamHasActivePlan();
     if (isPlanActive) {
-      return true;
+      return { isActive: true, isTrial: false };
+    }
+  }
+
+  // check if user has at least on membership with an trial plan
+  for (const team of teams) {
+    const teamBillingService = new InternalTeamBilling(team);
+    const isTrial = await teamBillingService.checkIfTeamHasTrialPlan();
```

Comment:
I see we're making two calls to the same endpoint to Stripe to get the subscription information. Instead, could we refactor this so we only have to make one call?